include/deal.II/grid/grid_tools.h

```diff
@@ -2651,6 +2654,39 @@ namespace GridTools
     const std::vector<BoundingBox<spacedim>> &local_bboxes,
     MPI_Comm                                  mpi_communicator);
 
+  namespace bg  = boost::geometry;
+  namespace bgi = boost::geometry::index;
+
+  template <int spacedim>
+  using bgPoint = bg::model::point<double, spacedim, bg::cs::cartesian>;
+
+  template <int spacedim>
+  using bgBox = bg::model::box<bgPoint<spacedim>>;
+
+  template <int spacedim, int leaf_number>
+  using LinearPointRtree = bgi::rtree<std::tuple<bgBox<spacedim>, unsigned int>,
+                                      bgi::linear<leaf_number>>;
+
+  /**
+   * Compute a bounding box tree which describes all locally owned portions of
+   * a distributed triangulation.
+   *
+   * Given a vector of bounding boxes describing the portion of the
+   * triangulation covered by locally owned cells and the relative
+   * parallel::distributed::Triangulation , compute the
+   * linear r-tree, used for search.
+   *
+   * The tree nodes contain tuples of two elements: the first being a bounding
+   * box, the second being the rank of the process whose local description
+   * contains the bounding box.
+   */
+  template <int dim, int spacedim, int leaf_number>
+  std::tuple<LinearPointRtree<spacedim, leaf_number>,
+             std::vector<std::tuple<bgBox<spacedim>, unsigned int>>>
```

Comment:
same here. The tree should be templated on the box type, not on derived tuples/pairs/etc of the box.